tower_of_hanoi.rb

```diff
@@ -0,0 +1,123 @@
+require 'pry'
+
+#NTS: Need to implement user input checking.
+#Variable pegs generation (2D array)
+#Refactor
+
+def check_solution  peg_to_check
+  solution = [3, 2, 1]
+  if solution == peg_to_check
+    puts "YOU WIN!!!!"
+    return true
+  end
+  return false
+end
+
+def render (peg_one, peg_two, peg_three)
+  # This method will render the pegs top-down
+  # The first peg will be printed, followed by a tab, then second and so on
+  iterator = 2
+  puts  "Current Board:"
+  while (iterator >= 0)
+    if (peg_one[iterator]!=nil)
+      print "o" * peg_one[iterator]
+    end
+    print "\t"
+    if (peg_two[iterator]!=nil)
+      print "o" * peg_two[iterator]
+    end
+    print "\t"
+    if (peg_three[iterator]!=nil)
+      print "o" * peg_three[iterator]
+    end
+    puts ""
+    iterator -= 1
+  end
+  puts "1\t2\t3\n"
+
+end
+
+def make_a_move (chosen_position, move_position, peg_one, peg_two, peg_three)
+  #NTS: refactor
+  case chosen_position
+  when 1
+    #Pop from peg_one onto 2 and 3
+    if((move_position == 2)&& (check_valid_move(peg_one,peg_two)))
+      peg_two.push(peg_one.pop)
+    elsif((move_position == 3)&& (check_valid_move(peg_one,peg_three)))
+      peg_three.push(peg_one.pop)
+    end
+  when 2
+    #Pop from peg_two onto 1 and 3
+    if((move_position == 1)&& (check_valid_move(peg_two,peg_one)))
+      peg_one.push(peg_two.pop)
+    elsif((move_position == 3)&& (check_valid_move(peg_two,peg_three)))
+      peg_three.push(peg_two.pop)
+    end
+  when 3
+    #Pop from peg_three onto 1 and 2
+    if((move_position == 1)&& (check_valid_move(peg_three,peg_one)))
+      peg_one.push(peg_three.pop)
+    elsif((move_position == 2)&& (check_valid_move(peg_three,peg_two)))
+      peg_two.push(peg_three.pop)
+    end
+  else
+    #NTS check
+    puts "invalid position"
+  end
+  puts "You moved a plate from #{chosen_position} to #{move_position}."
+end
+
+def check_valid_move (first_peg, second_peg)
+  if (second_peg[-1] == nil)
+    return true
+  end
+  if (second_peg[-1] < first_peg[-1])
+    puts ""
+    puts "ERROR: INVALID MOVE"
+    puts ""
+    return false
+  end
+  return true
+end
+
+def play_tower_of_hanoi
+  #Define pegs, which plates will be pushed onto. Peg one is leftmost
+  peg_one = [3, 2, 1]
+  peg_two = []
+  peg_three = []
+  player_movement  = []
+
+  player_quits = false
+  game_finished = false
+
+  puts "Welcome to Tower of Hanoi!"
+  puts "Instructions:"
+  puts "Enter where you'd like to move from and to"
+  puts "in the format [1,3] (without brackets). Enter 'q' to quit."
+  puts ""
+
+  until (player_quits || game_finished)
+    #NTS: refactor
+    render(peg_one, peg_two, peg_three)
+    print("Enter your move: ")
+    player_movement = gets.chomp.split(",")
+    player_movement = player_movement.map(&:to_i) #Converts array of strings to integers
+
+    if (player_movement[0] == "q")
+      player_quits = true
+      puts "You quit the game."
+      break
+    end
```

Comment:
"q" wouldn't quit the game for me. I got stuck in a loop where it just kept prompting me for another move. I think you want `exit` instead of `break` on line 110.